browser/resources/settings/brave_rewards_page/brave_rewards_page.ts

```diff
@@ -126,42 +154,121 @@ class SettingsBraveRewardsPage extends SettingsBraveRewardsPageBase {
       prefs.forEach((pref) => {
         if (pref.key === 'brave.brave_ads.should_allow_ads_subdivision_targeting') {
           this.getAdsDataForSubdivisionTargeting_()
+        } else if (pref.key === 'brave.rewards.external_wallet_type') {
+          this.isAutoContributeSupported_()
+          this.isInlineTippingSupported_()
         }
       }, this)
     })
   }
 
-  isAutoContributeSupported_() {
+  private openRewardsPanel_ () {
+    chrome.braveRewards.openRewardsPanel()
+    this.isAutoContributeSupported_()
+    this.isInlineTippingSupported_()
+  }
+
+  private splitVersion_ (version: string) {
+    return (version || 'invalid').split(/\./g).map((part) => {
+      const n = Number(part || 'invalid')
+      if (isNaN(n) || n < 0) {
+        throw new Error(`Invalid version string "${version}"`)
+      }
+      return n
+    })
+  }
+
+  // Compares two version strings. Returns 0 if the version strings refer to the
+  // same version, a positive number if the first version is "later", and a
+  // negative number if the first version is "earlier". Adapted from
+  // `base::Version::CompareTo`
+  private compareVersionStrings_ (version1: string, version2: string) {
```

Comment:
I'm guessing that we're not allowed to bring in the helper module over in `components/brave_rewards/resources/shared`?

Reply:
It seemed like it would cause a layering violation, but I'm open to other approaches to avoid duplicating this if possible.